backend/dataall/modules/datasets/services/dataset_service.py

```diff
@@ -623,3 +619,82 @@ def count_dataset_tables(session, dataset_uri):
             .filter(DatasetTable.datasetUri == dataset_uri)
             .count()
         )
+
+    @staticmethod
+    def query_environment_group_datasets(session, envUri, groupUri, filter) -> Query:
```

Comment:
These query functions should be in dataset_repository (following your design pattern), but as we discussed this could also be in a followup :)

Reply:
Yes, I will extract this methods after migrating the sharing.